helm-chart/renku/templates/_helpers.tpl

```diff
@@ -103,3 +103,47 @@ fail "External PostgreSQL and Renku-bundled PostgreSQL cannot both be disabled.
 {{- if .Values.global.externalServices.postgresql.enabled and .Values.global.externalServices.postgresql.password and .Values.global.externalServices.postgresql.existingSecret -}}
 fail "External PostgreSQL password and existing Secret fields cannot both be populated."
 {{- end -}}
+
+{{- define "keycloak.admin-secret" -}}
+{{- $secretAdmin := lookup "v1" "Secret" .Release.Namespace "keycloak-password-secret" -}}
+{{- if $secretAdmin -}}
+{{- if $secretAdmin.data.KEYCLOAK_ADMIN -}}
```

Comment:
I think this will crash if there is no `KEYCLOAK_ADMIN` in the secret? I think we should use `get` or something similar to aboid an error if the field is not set. I assume the purpose of the if statement is to avoid crashing if the field is not present in the secret.

Reply:
No, it doesn't crash, it return `false`. The purpose of the `if` statement on line 110 is to decide whether the secret exist with the var name expected from Keycloakx `KEYCLOAK_ADMIN`, which means that the migration already happened and the secret just needs to remain the same or (this is the else) if the secret has the var name used with Keycloak `KEYCLOAK_USER`.

If the secret does not exist at all (outer `if` nest) then the secrets are generated.